### PR TITLE
Add a linear dm device on the top of a nvme device

### PR DESCRIPTION
**Pull request by derekbit**
https://github.com/longhorn/longhorn/issues/7357

deps on https://github.com/longhorn/go-spdk-helper/pull/55

### CODE REVIEW

**Comment by innobead** [comment]
@derekbit there were some PRs merged. Please help rebase this PR. Thanks.

**Comment by derekbit** [comment]
> @derekbit there were some PRs merged. Please help rebase this PR. Thanks.

Updated and ready for review. cc @innobead @shuo-wu

File: pkg/spdk/server.go [hunk 1]
```diff
@@ -296,11 +316,16 @@ func (s *Server) ReplicaCreate(ctx context.Context, req *spdkrpc.ReplicaCreateRe
 	}
 
 	s.Lock()
+	spdkClient := s.spdkClient
 	if _, ok := s.replicaMap[req.Name]; !ok {
+		ready, err := s.checkLvsReadiness(req.LvsUuid, req.LvsName)
+		if err != nil || !ready {
+			s.Unlock()
```
Comment (innobead) [inline pkg/spdk/server.go:323]: Can we just move `s.Unlock()` to just next to `s.Lock()`, and then remove the following original `s.Unlock()`?

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just one comment.

File: pkg/spdk/server.go (hunk 1 above)
Comment (derekbit) [reply pkg/spdk/server.go]: Addressed. Thank you.

**Review by innobead — APPROVED** [review]

**Pull request merged**